src/autoscaler/api/brokerserver/broker_handler.go

```diff
@@ -179,33 +249,20 @@ func (h *BrokerHandler) BindServiceInstance(w http.ResponseWriter, r *http.Reque
 
 		h.logger.Info("creating/updating schedules", lager.Data{"policy": body.Policy})
 		err = h.schedulerUtil.CreateOrUpdateSchedule(body.AppID, body.Policy, policyGuid.String())
+		//while there is synchronization between policy and schedule, so creating schedule error does not break
+		//the whole creating binding process
 		if err != nil {
-			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
-				Code:    "Interal-Server-Error",
-				Message: "Error creating/updating schedules"})
-			return
+			h.logger.Error("failed to create/update schedules", err, lager.Data{"policy": body.Policy})
 		}
 	}
-
-	err = h.bindingdb.CreateServiceBinding(bindingId, instanceId, body.AppID)
-	if err != nil {
-		//revert policy saving when the create service binding failed
-		errDeletePolicy := h.policydb.DeletePolicy(body.AppID)
-		if errDeletePolicy != nil {
-			h.logger.Error("failed to delete policy after binding failure", errDeletePolicy, lager.Data{"bindingErr": err, "errDeletePolicy": errDeletePolicy})
-		}
-
-		if err == db.ErrAlreadyExists {
-			w.Write(nil)
-			return
-		}
-		handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
-			Code:    "Interal-Server-Error",
-			Message: "Error creating service binding"})
-		return
-	}
-	w.WriteHeader(http.StatusCreated)
-	w.Write(nil)
+	handlers.WriteJSONResponse(w, http.StatusCreated, models.CredentialResponse{
+		Credentials: models.Credentials{
+			CustomMetrics: models.CustomMetrics{
+				CustomMetricCredentials: cred,
+				URL:                     h.conf.MetricsForwarder.MetricsForwarderUrl,
+			},
+		},
```

Comment:
@boyang9527   @silvestre , could you share your insight here ?    The current binding credential format  looks complex ...
```
"credentials":{
  "custom_metrics": {
     "username": xxx,
     "password": xxx,
     "url:xxx
  }
}
```

Reply:
If possible, this should not change, as we already have that API GA for our customers. If there's a change here we would need maintain the old structure in our fork.

Reply:
ok， then let us keep it as is